runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -105,6 +105,18 @@
     @Getter(lazy = true)
     private final long snapshotTimestamp = obtainSnapshotTimestamp();
 
+
+    /**
+     * these hints are initialized when the transaction retrieves the snapshotTimestamp.
+     * These hints are used service to locally service stream tail queries instead of
+     * making a request to the sequencer.
+     */
+    @Getter
+    public Map<UUID, Long> sequencerHints = null;
+
+    @Getter
+    public long hintsEpoch;
```

Comment:
In addition, this field seems to be not useful, the epoch is only used on a write, there's no need to actually keep this?

Reply:
removed.